src/ray/core_worker/store_provider/plasma_store_provider.cc

```diff
@@ -16,72 +16,101 @@ Status CoreWorkerPlasmaStoreProvider::Put(const RayObject &object,
   return local_store_provider_.Put(object, object_id);
 }
 
+Status CoreWorkerPlasmaStoreProvider::Create(const std::shared_ptr<Buffer> &metadata,
+		                               const size_t data_size,
+                                               const ObjectID &object_id,
+					       std::shared_ptr<Buffer> &data) {
+  return local_store_provider_.Create(metadata, data_size, object_id, data);
+}
+
+Status CoreWorkerPlasmaStoreProvider::Seal(const ObjectID &object_id) {
+  return local_store_provider_.Seal(object_id);
+}
+
 Status CoreWorkerPlasmaStoreProvider::Get(
     const std::vector<ObjectID> &ids, int64_t timeout_ms, const TaskID &task_id,
     std::vector<std::shared_ptr<RayObject>> *results) {
   (*results).resize(ids.size(), nullptr);
-
-  bool was_blocked = false;
-
-  std::unordered_map<ObjectID, int> unready;
+  RAY_CHECK_OK(raylet_client_->FetchOrReconstruct(ids, /*fetch_only=*/true, task_id));
```

Comment:
I think it's better to also put the first batch in a loop, so we can also batch the requests.

Reply:
Also, the changes of this function looks orthogonal to Python. Better use a separate PR.

Reply:
Ah you're right, we should also be batching the `fetch_only=true` requests.

I suppose the changes are orthogonal, but they're a precondition for merging this PR. I think it would be pretty high overhead for the benefit to split it off, but I can if you insist on it.

Reply:
One easy way to split c++ changes out could be:
```
# use a different branch name.
git check -b another_branch
# reset irrelevant files to master.
git checkout master -- python
git checkout master -- java
..
```
I didn't test this. If it turns out to be hard to do so, it's okay to leave it as is as well.